src/client/datascience/jupyter/interpreter/jupyterInterpreterDependencyService.ts

```diff
@@ -128,15 +128,28 @@ export class JupyterInterpreterDependencyService {
         _error?: JupyterInstallError,
         token?: CancellationToken
     ): Promise<JupyterInterpreterDependencyResponse> {
-        const missingProducts = await this.getDependenciesNotInstalled(interpreter, token);
+        // If we're dealing with a non-conda environment & pip isn't installed, we can't install anything.
+        // Hence prompt to install pip as well.
+        const pipInstalledInNonCondaEnvPromise =
+            interpreter.envType === EnvironmentType.Conda
+                ? Promise.resolve(undefined)
+                : this.installer.isInstalled(Product.pip, interpreter);
+
+        const [missingProducts, pipInstalledInNonCondaEnv] = await Promise.all([
+            this.getDependenciesNotInstalled(interpreter, token),
+            pipInstalledInNonCondaEnvPromise
+        ]);
         if (Cancellation.isCanceled(token)) {
             return JupyterInterpreterDependencyResponse.cancel;
         }
         if (missingProducts.length === 0) {
             return JupyterInterpreterDependencyResponse.ok;
         }
 
-        const message = getMessageForLibrariesNotInstalled(missingProducts, interpreter.displayName);
+        const message = getMessageForLibrariesNotInstalled(
+            pipInstalledInNonCondaEnv ? [Product.pip].concat(missingProducts) : missingProducts,
```

Comment:
This seems backwards?

If ```pipInstallaedInNonCondaEnv``` is true, doesn't it add 'pip' to the list of stuff to install?

Reply:
Fixed, thanks